app/src/main/java/com/vitaliidiadchenko/mykotkinapplication/FragmentMovieList.kt

```diff
@@ -47,17 +52,24 @@ class FragmentMovieList : Fragment() {
 
     private val movieListener = object : OnPosterCardClickListener {
         override fun onClick(movie: Movie) {
-            listener?.goToFragmentMoviesDetails()
+            listener?.goToFragmentMoviesDetails(movie)
+        }
+    }
+
+    private fun setMovies() {
+        scope.launch {
+            movies = context?.let { loadMovies(it)}
+            updateData()
         }
     }
 
     private fun updateData() {
```

Comment:
Список фильмов можно передать параметром функции и избавиться от свойства `private var movies: List<Movie>? = null` в классе
```suggestion
    private fun updateData(movies: List<Movie>) {
```